aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -370,44 +300,36 @@ Tensor fft_ifftn(const Tensor& self, c10::optional<IntArrayRef> s,
 
 Tensor fft_rfftn(const Tensor& self, c10::optional<IntArrayRef> s,
                 c10::optional<IntArrayRef> dim,
-                c10::optional<std::string> norm) {
+                c10::optional<std::string> norm_str) {
+  TORCH_CHECK(!self.is_complex(), "Expected a real input tensor to rfftn");
```

Comment:
"Expected a real input tensor to rfftn" -> "rfftn expects a real-valued input tensor, but got {dtype}!"